xdsmatcher/pkg/matcher/types/types.go

```diff
@@ -0,0 +1,33 @@
+package types
```

Comment:
Maybe some comments in this file above the important types/interfaces might be useful? Like the `Matcher` interface for example. Just so it shows up nicely in godoc for people consuming the framework